website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+
+### The Faithfulness Metric
+
+As Worldwide WanderAgency's AI capabilities grow, they decide to implement an AI-powered customer service chatbot to handle inquiries in multiple languages.
+
+After making sure their materials are well translated.
+The company decided to build a pipline for their materials which is invoked inside a `RAG` system.
+
+ To ensure the chatbot provides accurate information based on the company's policies, they create another custom metric: the `"faithfulness"` metric.
+
+This metric assesses how well the chatbot's responses align with the provided context (company policies) and the specific questions asked by customers.
+
+### Built in Metrics
+
+In addition to this we can leverage mlflow built in metrics to measure some relevant metrics.
+Since Worldwide WanderAgency uses a text model we get these metrics out of the box. 
+
+- [toxicity](https://huggingface.co/spaces/evaluate-measurement/toxicity) 
+- [ari_grade_level](https://en.wikipedia.org/wiki/Automated_readability_index) 
+- [flesch_kincaid_grade_level](https://en.wikipedia.org/wiki/Flesch%E2%80%93Kincaid_readability_tests#Flesch%E2%80%93Kincaid_grade_level) 
+
+For this use case we want to make sure none of the context translated are toxic.
+
+
+## Putting It All Together: Evaluating Worldwide WanderAgency's AI Systems
+
+With these custom metrics in place, the travel agency can now use MLflow Evaluate to assess both their translation model and customer service chatbot. They can track performance over time, identify areas for improvement, and ensure that their AI-driven communication maintains high standards of cultural sensitivity and accuracy.
+
+Here's a glimpse of how they might visualize their evaluation results:
+
+![Gauge Chart](gauge.png)
+
+This dashboard allows Worldwide WanderAgency's team to quickly assess the performance of their AI systems across different metrics, ensuring they're meeting the high standards required for effective global communication.
+
+To demonstrate the power of using LLMs as judges with MLflow evaluate, let's walk through an example of evaluating a language translation model. We'll create a custom metric called "cultural_sensitivity" to assess how well our translation model preserves cultural nuances. 
+
+For the full code example, please refer to the accompanying notebook. Here's a brief overview of the process:
+
+start by installing all the necessary libraries for this demo to work.
+Since mlflow evaluate offers some default metrics we need to install the libraries to calculate those metrics (toxiticy etc..)
+
+```python
+!pip install mlflow==2.14.1 openai  transformers torch torchvision evaluate datasets openai tiktoken fastapi rouge_score textstat tenacity plotly ipykernel nbformat==5.10.4
+
+!pip install --upgrade nbformat
+```
+
+
+We will be using gpt3.5 and gpt4 during this example for that let's start by setting up an open ai key
```

Comment:
- open ai -> OpenAI
- general note throughout section: be consistent with punctuation e.g. end instructions before a code snippet with a colon or period